cmd/kubelet/app/options/options.go

```diff
@@ -466,8 +466,10 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.BoolVar(&c.MakeIPTablesUtilChains, "make-iptables-util-chains", c.MakeIPTablesUtilChains, "If true, kubelet will ensure iptables utility rules are present on host.")
 	fs.Int32Var(&c.IPTablesMasqueradeBit, "iptables-masquerade-bit", c.IPTablesMasqueradeBit, "The bit of the fwmark space to mark packets for SNAT. Must be within the range [0, 31]. Please match this parameter with corresponding parameter in kube-proxy.")
 	fs.Int32Var(&c.IPTablesDropBit, "iptables-drop-bit", c.IPTablesDropBit, "The bit of the fwmark space to mark packets for dropping. Must be within the range [0, 31].")
-	fs.StringVar(&c.ContainerLogMaxSize, "container-log-max-size", c.ContainerLogMaxSize, "<Warning: Beta feature> Set the maximum size (e.g. 10Mi) of container log file before it is rotated.")
-	fs.Int32Var(&c.ContainerLogMaxFiles, "container-log-max-files", c.ContainerLogMaxFiles, "<Warning: Beta feature> Set the maximum number of container log files that can be present for a container. The number must be >= 2.")
+	fs.StringVar(&c.ContainerLogMaxSize, "container-log-max-size", c.ContainerLogMaxSize, "<Warning: Beta feature> Set the maximum size (e.g. 10Mi) of container log file before it is rotated. This flag can only be used with --container-runtime=remote.")
+	fs.Int32Var(&c.ContainerLogMaxFiles, "container-log-max-files", c.ContainerLogMaxFiles, "<Warning: Beta feature> Set the maximum number of container log files that can be present for a container. The number must be >= 2. This flag can only be used with --container-runtime=remote.")
+	fs.Int32Var(&c.ContainerLogMaxWorkers, "container-log-max-workers", c.ContainerLogMaxWorkers, "<Warning: Beta feature> Set the maximum number of concurrent worker to spawn for performing the log rotation. Set the value to 0 to disable concurrent rotation. Defaults to 0")
+	fs.DurationVar(&c.ContainerLogMonitorPeriod.Duration, "container-log-monitor-period", c.ContainerLogMonitorPeriod.Duration, "<Warning: Beta feature> Set the duration at which kubelet will monitor the logs for performing rotation. Defaults to 10s.")
```

Comment:
Both are added as "Beta features". What do we mean by "Beta features"?
The existing `containerLogMaxSize` and `containerLogMaxFiles` could have been GA'ed since the feature gate behind it has been GA'ed and removed.

Reply:
@harshanarayana I know this PR has turned into a marathon, but any chance you able to rebase and respond here? Would love to help get this over the line ❤️ If so, I can bring it up in the sig-node meeting to see if we can close the loop on this.

Reply:
@mmerkes  Yes. Definitely. I will do that today.

Reply:
@tengqm 

The warning is added to the new flags added. 

> The existing containerLogMaxSize and containerLogMaxFiles could have been GA'ed since the feature gate behind it has been GA'ed and removed.

The warning exists for these arguments in the mainline branch today as well. It has not been modified in this PR. The newly added arguments are not really backed by a feature gate as suggested by @dims  in the PR conversation but I wanted to indicate that they are not fully GA'ed per say.

Reply:
> @harshanarayana I know this PR has turned into a marathon, but any chance you able to rebase and respond here? Would love to help get this over the line ❤️ If so, I can bring it up in the sig-node meeting to see if we can close the loop on this.

@mmerkes When you are planning to bring this up in the sig-node meeting, please give me a heads up. I can join the call and we can sort out anything that comes up in that discussion quickly. I will drop you a note on the slack.

Reply:
> The warning exists for these arguments in the mainline branch today as well. It has not been modified in this PR. The newly added arguments are not really backed by a feature gate as suggested by @dims in the PR conversation but I wanted to indicate that they are not fully GA'ed per say.

@harshanarayana can you please confirm that if the 2 new config params are not specified the behavior will look and feel like exactly what the current code (before this PR) does?

Reply:
Yes. That is correct.  The default values for these two params are set so that they match the current behavior. 

```go
	for i := 0; i <= c.maxWorkers; i++ {
		worker := i + 1
		go c.processQueueItems(ctx, worker)
	}
```
This will only spin up one go routine worker for cleanup since the default value for the worker is set to 0.

This will then start picking up one entry from the queue at a time and process like it would have done in the previous case. The wait loop in the `rotateLogs` handler ensures that no new entry is pushed to the queue until the current queue is empty to make sure it gives the same behavior as the current one would when iterating over the list of pods it found. There has been no change in logic under `processContainer` other than moving some call to that function from inside the `rotateLogs` handler.  @dims  I hope that answers the question ?

Do you want me to run any other test to confirm this again? I can do that if required.

Reply:
all good. thanks @harshanarayana